controllers/shooter.py

```diff
@@ -27,65 +27,85 @@ class ShooterController(StateMachine):
 
     def __init__(self) -> None:
         # set default preference for goal height
-        pass
+        self.GoalHeight_Preference = GoalHeight.HIGH
 
     @state(first=True, must_finish=True)
     def preparing_intake(self) -> None:
         # send pitch and turret to angle
-
+        self.turret_component.set_angle(0.0)
         # if at pitch and turret angle
+        if self.turret_component.at_angle():
+            self.next_state("intaking")
         # next state intaking
-        pass
+        #pass
 
     @state(must_finish=True)
     def intaking(self) -> None:
-        # shooter intake
+        # shooter intake 
         # intake deploy
+        self.intake_component.deploy()
 
-        # if shooter is loaded
-        # retract intake
-        # next state tracking
-        pass
+        if self.shooter_component.is_loaded():
+            self.intake_component.retract()
+            self.next_state("tracking")
+        #pass
 
     @default_state
     def tracking(self) -> None:
         # if turret not indexed
         # return
+        if self.turret_component.find_index() is None:
+            return
 
         # update flywheel speed
+        self.shooter_component.set_flywheel_speed(1.0)
         # update turret angle
+        self.turret_component.set_angle(0.0)
         # update tilt angle
-
-        pass
+        self.tilt_component.set_angle(0.0)
+        #pass
 
     @state(must_finish=True)
     def shooting(self) -> None:
         # update flywheel speed
+        self.shooter_component.set_flywheel_speed(1.0)
         # update turret angle
+        self.turret_component.set_angle()
         # update tilt angle
+        self.tilt_component.set_angle()
 
         # if tilt ready and turret ready and flywheels at speed
         # shoot
         # next state tracking
-        pass
+        if self.tilt_component.at_angle() and self.turret_component.at_angle(): #and self.shooter_component.at_flywheel_speed():
+            self.shooter_component.shoot()
+            self.next_state("tracking")
```

Comment:
After the cube is gone, we need to reset everything so that we are ready to intake again.
Tracking is not the correct next state.